provider/oauth2.go

```diff
@@ -4,30 +4,39 @@ import (
 	"context"
 	"encoding/json"
 	"fmt"
+	"github.com/MicahParks/keyfunc"
+	"github.com/pkg/errors"
 	"io"
 	"net/http"
 	"strings"
+	"sync"
 	"time"
 
 	"github.com/go-pkgz/rest"
 	"github.com/golang-jwt/jwt"
+	jwtv4 "github.com/golang-jwt/jwt/v4"
 	"golang.org/x/oauth2"
 
 	"github.com/go-pkgz/auth/logger"
 	"github.com/go-pkgz/auth/token"
 )
 
+const clockSkew = 10 * time.Second
+
 // Oauth2Handler implements /login, /callback and /logout handlers from aouth2 flow
 type Oauth2Handler struct {
 	Params
 
 	// all of these fields specific to particular oauth2 provider
 	name     string
 	infoURL  string
+	jwksURL  string
 	endpoint oauth2.Endpoint
 	scopes   []string
 	mapUser  func(UserData, []byte) token.User // map info from InfoURL to User
 	conf     oauth2.Config
+	keyfunc  jwt.Keyfunc
```

Comment:
maybe make the relation between these to more explicit with internal struct{fn,mutex}?